code/renderers/svelte/src/docs/extractArgTypes.ts

```diff
@@ -102,25 +102,24 @@ const parseTypeToControl = (type: JSDocType | undefined): any => {
   if (type.kind === 'type') {
     switch (type.type) {
       case 'string':
-        return { type: 'text' };
-
-      case 'enum':
-        return { type: 'radio' };
+        return { control: { type: 'text' } };
       case 'any':
-        return { type: 'object' };
+        return { control: { type: 'object' } };
       default:
-        return { type: type.type };
+        return { control: { type: type.type } };
     }
   } else if (type.kind === 'union') {
-    // @ts-expect-error TODO: fix, this seems like a broke in package update
-    if (Array.isArray(type.type) && !type.type.find((t) => t.type !== 'string')) {
+    if (
+      Array.isArray(type.type) &&
+      !type.type.find((t) => t.kind !== 'const' || t.type !== 'string')
+    ) {
```

Comment:
notice that unions of numbers or a mix of numbers and strings don't get parsed into a radio, only pure string unions does

![image](https://github.com/user-attachments/assets/5671fca7-0c80-427b-bc89-8e3c8a583d00)
![image](https://github.com/user-attachments/assets/588a94c7-97e5-437d-8cad-25801366265e)

it looks like with a simple change of widening the condition above to also allow numbers, undefined and null:

```js
    if (
      Array.isArray(type.type) &&
      !type.type.some((t) => t.kind !== 'const' || ['string', 'number', 'null', 'undefined'].includes(t.type)
   ) {
```

and then using `String(value)` for the labels:

```js
const options = type.type.map((t) => (t as JSDocTypeConst).value);

return {
  control: {
    type: 'radio',
    labels: options.map(String)
  },
  options
}
```

We could make the union support way more powerful with a simple change.

this is also sort of unrelated to the work here, so feel free to ignore it for now if you want to.

Reply:
great!